vmm/scripts/kernel/build-kernel/how-to-tailor-linux-kernel-for-kuasar-security-container.md

```diff
@@ -0,0 +1,209 @@
+This document describes the tailoring and building methods of the linux kernel for the kuasar security container in different scenarios. Developers can quickly and automatically build a linux kernel image which adapts to the kuasar security container  by using the provided `build-kernel.sh` script.
+
+## Tailoring Method
+
+Linux kernel tailoring is a process of removing or retaining some kernel features or modules based on actual application scenario requirements to achieve the purposes of optimizing system performance, reducing memory usage, and improving security. 
+Generally, there are two methods to tailor the linux kernel:
+
+1.  **Subtraction Tailoring**: Disable the configuration options of unnecessary features based on the default kernel configuration.
+2.  **Addition Tailoring**: Developers know which kernel capabilities are needed, and combine the various kernel feature configuration options from scratch.
+
+The two tailoring methods have their own advantages and disadvantages:
+
+ *  **The advantage of the "subtraction" tailoring method is that it is simple and convenient**, and can quickly tailor the kernel to meet functional requirements. However, the disadvantage is that every time the kernel version is updated, **the manual tailoring process needs to be repeated, and the tailoring process cannot be automated and inherited**.
+ *  **The advantage of the "addition" tailoring mode is that the kernel can quickly and automatically tailored for different versions of the kernel**, and the memory footprint of the tailored kernel is very small. **The disadvantage is that it is difficult to get started**. Developers need to be familiar with the features of the kernel and know how to divide the kernel configuration options according to the kernel features.**This may require a significant amount of time for the first tailoring.**
```

Comment:
```suggestion
 *  **The advantage of the "addition" tailoring mode is that the kernel can quickly and automatically tailored for different versions of the kernel**, and the memory footprint of the tailored kernel is very small. **The disadvantage is that it is difficult to get started**. Developers need to be familiar with the features of the kernel and know how to divide the kernel configuration options according to the kernel features. **This may require a significant amount of time for the first tailoring.**
```
Typo